Documents/TechnicalSpecifications/technicalSpecifications.md

```diff
@@ -162,4 +211,275 @@ Whitespace<br>
+	if new_closest != cached_closest:
+		if is_instance_valid(cached_closest):
+			unfocus(cached_closest)
+		if new_closest:
+			focus(new_closest)
+		cached_closest = new_closest
+
+# Handle area exit events
+func _on_area_exited(area: Interactable) -> void:
+	if cached_closest == area:
+		unfocus(area)
+```
+
+Then when the player tries to interact with the object:
+
+```GDscript
+# Handle interaction inputs
+func _input(event: InputEvent) -> void:
+	if event.is_action_pressed("interact"):
+		if cached_closest:
+			# Check if interacting with a Sheet
+			if cached_closest.get_parent() is Sheet:
+				interact_with_sheet(cached_closest.get_parent())
+			elif held_object:
+				# If holding an object, interact with the bin
+				if cached_closest.get_parent() is Bin:
+					interact_with_bin(cached_closest.get_parent())
+				else:
+					# Release the held object
+					release_held_object()
+			else:
+				# Interact with an object, a bin, the teleporter
+				if cached_closest.get_parent() is Bin:
+					interact_with_bin(cached_closest.get_parent())
+				elif cached_closest.get_parent() is RigidBody3D:
+					grab_object(cached_closest.get_parent() as RigidBody3D)
+				elif cached_closest.get_parent() is Teleporter:
+					interact_with_teleporter(cached_closest.get_parent())
+
+# Grab a new object
+func grab_object(object: RigidBody3D) -> void:
+	if object:
+		# Release the currently held object first (if any)
+		release_held_object()
+		# Grab the new object
+		object.hold(player)
+		held_object = object
+```
+
+Also, if the player already has an object, he drops it when trying to take an other object or when he presses the interaction button with nothing around:
+
+```GDscript
+# Release the currently held object
+func release_held_object() -> void:
+	if held_object:
+		held_object.place_on_ground()
+		held_object.release()
+		held_object = null
+```
+
+The player will not be able to hold more than one item at a time.
+
+### Other interactions
+
+Same for the object interactions, the player will have to be close enough to what can be interacted with. The distance detection is the same, and the interaction action is shared with the one for the objects while being adapted for the correct interaction.
+
+### Call
+
+The player will be able to call the scientist to get information on things needed, if he has unlocked what he can have information on. To do so, the player will have to press the 'C' button on the keyboard.
+
+### Changing zone
+
+While there are multiple different zones, the player has to be able to go to each zone. To do so, the player has to interact with a teleporter. Then the player will get teleported to the chosen zone:
+
+```GDscript
+func _on_interactable_interacted(interactor: Interactor) -> void:
+	add_highlight()
+	#need to ask confirmation to the player
+	get_tree().change_scene_to_file("[destination]")
+```
+
+On the last line, [destination] has to be changed depending on where the player is.
+
+# Risks and mitigation strategies
```

Comment:
What is the impact of the risks and it's frequency.

I recommend to rework this section to include the impact and the likelihood

Reply:
don't worry :)